test/fork/migration-v2.js

```diff
@@ -555,6 +638,101 @@ describe('v2 migration', function () {
+        minAmount: assetsData[i].minAmount,
+        maxAmount: assetsData[i].maxAmount,
+        lastSwapTime: assetsData[i].lastSwapTime,
+        maxSlippageRatio: assetsData[i].maxSlippageRatio,
+      };
+      return acc;
+    }, {});
+    this.contractData.pool.after.minPoolEth = await this.swapOperator.minPoolEth();
+    expect(this.contractData.pool.after.minPoolEth, "AssertionError: values of minPoolEth don't match\n").to.be.equal(
+      this.contractData.pool.before.minPoolEth,
+    );
+
+    const DENOMINATOR_DIFFERANCE = Math.pow(10, 14);
+    Object.entries(this.contractData.pool.before.assetsData).forEach(([asset, value]) => {
+      expect(
+        this.contractData.pool.after.assetsData[asset].minAmount,
+        `AssertionError: values of minAmount in ${asset} don't match\n`,
+      ).to.be.equal(value.minAmount);
+
+      expect(
+        this.contractData.pool.after.assetsData[asset].maxAmount,
+        `AssertionError: values of maxAmount in ${asset} don't match\n`,
+      ).to.be.equal(value.maxAmount);
+
+      expect(
+        this.contractData.pool.after.assetsData[asset].lastSwapTime,
+        `AssertionError: values of lastSwapTime in ${asset} don't match\n`,
+      ).to.be.oneOf([value.lastSwapTime, 0]);
+
+      expect(
+        this.contractData.pool.after.assetsData[asset].maxSlippageRatio,
+        `AssertionError: values of maxSlippageRatio in ${asset} don't match\n`,
+      ).to.be.equal(value.maxSlippageRatio.div(DENOMINATOR_DIFFERANCE));
+    });
+
+    // PRICE FEED
+    const assetsEthRate = await Promise.all(
+      this.contractData.pool.before.assets.map(address => this.priceFeedOracle.getAssetToEthRate(address)),
+    );
+    const getAssetForEth = await Promise.all(
+      this.contractData.pool.before.assets.map(address => this.priceFeedOracle.getAssetForEth(address, 10)),
+    );
+
+    this.contractData.priceFeedOracle.after.assetsEthRate = this.contractData.pool.before.assets.reduce(
+      (acc, asset, i) => {
+        acc[asset] = assetsEthRate[i];
+        return acc;
+      },
+      {},
+    );
+    this.contractData.priceFeedOracle.after.assetsForEth = this.contractData.pool.before.assets.reduce(
+      (acc, asset, i) => {
+        acc[asset] = getAssetForEth[i];
+        return acc;
+      },
+      {},
+    );
+
+    Object.entries(this.contractData.priceFeedOracle.before.assetsEthRate).forEach(([asset, value]) => {
+      expect(
+        this.contractData.priceFeedOracle.after.assetsEthRate[asset],
+        `AssertionError: values of assetsEthRate in ${asset} don't match\n`,
+      ).to.be.equal(value);
+    });
+
+    Object.entries(this.contractData.priceFeedOracle.before.assetsForEth).forEach(([asset, value]) => {
+      expect(
+        this.contractData.priceFeedOracle.after.assetsForEth[asset],
+        `AssertionError: values of assetsEthRate in ${asset} don't match\n`,
+      ).to.be.equal(value);
+    });
+  });
+
   it.skip('deploy staking pool', async function () {
```

Comment:
let's assert one more thing here post upgrades:

for each contract assert it's dependencies in MasterAwareV2.internalContracts are set correctly.

`internalContracts` is now made public so you can just read them.

Each contract has a subset of deps so we could whitelist those for each, an approach similar to all the generalized assertions done til now will work nicely.

If anyone thinks this should be a different Pr, i'm ok with that, but the complexity is not that high

we

Reply:
Probably better to implement this in the basic functionality tests.